File: src/main/java/com/uiptv/util/DeviceId2AttributeParser.java
package com.uiptv.util;

import java.text.Normalizer;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class DeviceId2AttributeParser implements StalkerAttributeParser {
    private static final Pattern DEVICE_ID2_PATTERN = Pattern.compile("(?:id\\s*2|device\\s*id\\s*2).*?([A-F0-9]{10,64})\\b", Pattern.CASE_INSENSITIVE);

    @Override
    public String parse(String line) {
        String normalizedLine = Normalizer.normalize(line, Normalizer.Form.NFKD);
        Matcher matcher = DEVICE_ID2_PATTERN.matcher(normalizedLine);
        if (matcher.find()) {
            return matcher.group(1);
        }
        return null;
    }

    @Override
    public StalkerAttributeType getAttributeType() {
        return StalkerAttributeType.DEVICE_ID_2;
    }
}

File: server/src/main/kotlin/com/uiptv/util/DeviceId2AttributeParser.kt
package com.uiptv.util

import com.uiptv.api.StalkerAttributeParser
import java.text.Normalizer
import java.util.regex.Pattern

class DeviceId2AttributeParser : StalkerAttributeParser {
    override fun parse(line: String): String? {
        val normalizedLine = Normalizer.normalize(line, Normalizer.Form.NFKD)
        val matcher = DEVICE_ID2_PATTERN.matcher(normalizedLine)
        return if (matcher.find()) matcher.group(1) else null
    }

    override fun getAttributeType(): StalkerAttributeType = StalkerAttributeType.DEVICE_ID_2

    private companion object {
        val DEVICE_ID2_PATTERN: Pattern =
            Pattern.compile("(?:id\\s*2|device\\s*id\\s*2).*?([A-F0-9]{10,64})\\b", Pattern.CASE_INSENSITIVE)
    }
}
